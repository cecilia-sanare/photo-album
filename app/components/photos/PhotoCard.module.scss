.card {
    position: relative;
    display: flex;
    flex-direction: column;
    gap: 10px;
    padding: 10px;
    box-sizing: border-box;
    width: 150px;
    border-radius: 3px;
    background: rgba(0, 0, 0, 0.25);
    color: white;
    cursor: pointer;

    transition: 0.15s ease-in-out;
    transition-property: background-color;

    &:hover {
        background: rgba(0, 0, 0, 0.5);
    }

    > .thumbnail {
        border-radius: inherit;
    }

    > .id {
        display: flex;
        align-items: center;
        padding: 0 10px;
        min-height: 30px;
        line-height: 1;
        background: rgba(0, 0, 0, 0.5);
        border-radius: 3px;
        position: absolute;
        top: 15px;
        left: 15px;
    }
}