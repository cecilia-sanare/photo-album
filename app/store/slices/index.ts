import { combineReducers } from '@reduxjs/toolkit';
import photosSlice from './photos.slice';
import albumsSlice from './albums.slice';
import searchSlice from './search.slice';

export const reducers = combineReducers({
  albums: albumsSlice,
  photos: photosSlice,
  search: searchSlice
});
