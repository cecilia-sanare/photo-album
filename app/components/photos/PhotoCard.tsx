import { Photo } from '../../store/slices/photos.slice';
import * as styles from './PhotoCard.module.scss';

type PhotoCardProps = {
  photo: Photo;
  onClick?: () => void;
}

export function PhotoCard({
  photo,
  onClick
}: PhotoCardProps) {
  return (
    <div
      className={styles.card}
      data-testid='photo-card'
      onClick={onClick}
    >
      <div className={styles.id}>{photo.id}</div>
      <img className={styles.thumbnail} src={photo.thumbnailUrl} />
      {photo.title}
    </div>
  )
}
