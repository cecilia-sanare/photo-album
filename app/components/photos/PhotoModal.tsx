import { useCallback } from 'react';
import { Photo } from '../../store/slices/photos.slice'
import { setSelectedPhoto } from '../../store/slices/search.slice';
import { useAppDispatch } from '../../store/store';
import { Modal } from '../common/Modal'
import { ModalHeader } from '../common/ModalHeader';
import { ModalContent } from '../common/ModalContent';

type PhotoModalProps = {
  photo?: Photo;
}

export function PhotoModal({
  photo
}: PhotoModalProps) {
  const dispatch = useAppDispatch();

  const onClose = useCallback(() => {
    dispatch(setSelectedPhoto(null));
  }, [dispatch]);

  return (
    <Modal
      opened={Boolean(photo)}
      onClose={onClose}
    >
      <ModalHeader onClose={onClose}>
        {photo?.title}
      </ModalHeader>
      <ModalContent>
        <img src={photo?.url} />
      </ModalContent>
    </Modal>
  )
}
