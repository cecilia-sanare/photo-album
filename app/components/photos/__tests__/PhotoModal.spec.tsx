import { fireEvent, render } from '@testing-library/react';
import { PhotoModal } from '../PhotoModal';
import Chance from 'chance';
import { Photo } from '../../../store/slices/photos.slice';
import { useAppDispatch } from '../../../store/store';

jest.mock('../../../store/store');

const useAppDispatchMocked = jest.mocked(useAppDispatch);

const chance = new Chance();

describe('<PhotoModal />', () => {
  let dispatch;

  beforeEach(() => {
    dispatch = jest.fn();

    useAppDispatchMocked.mockReturnValue(dispatch);
  });

  it('should render the PhotoModal', () => {
    const photo: Photo = {
      albumId: chance.integer(),
      id: chance.integer(),
      thumbnailUrl: chance.url(),
      title: chance.word(),
      url: chance.url()
    };

    const component = render(<PhotoModal photo={photo} />);

    expect(component.baseElement).toMatchSnapshot();
  });

  it('should reset the selected photo when the modal is closed', async () => {
    const photo: Photo = {
      albumId: chance.integer(),
      id: chance.integer(),
      thumbnailUrl: chance.url(),
      title: chance.word(),
      url: chance.url()
    };

    const component = render(<PhotoModal photo={photo} />);

    expect(dispatch).toHaveBeenCalledTimes(0);

    await fireEvent.click(component.getByTestId('modal-close'));

    expect(dispatch).toHaveBeenCalledTimes(1);
  });

  describe('prop(photo)', () => {
    it('should support being undefined', () => {
      const component = render(<PhotoModal />);

      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
