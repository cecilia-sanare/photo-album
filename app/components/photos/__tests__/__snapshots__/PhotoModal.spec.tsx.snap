// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoModal /> prop(photo) should support being undefined 1`] = `
<body
  class=""
>
  <div>
    <div
      class="backdrop"
      data-testid="backdrop"
    />
    <div
      class="modal"
      data-testid="modal"
    >
      <div
        class="header"
      >
        <div
          class="close"
          data-testid="modal-close"
        >
          x
        </div>
      </div>
      <div
        class="content"
      >
        <img />
      </div>
    </div>
  </div>
</body>
`;

exports[`<PhotoModal /> should render the PhotoModal 1`] = `
<body
  class="no-scroll"
>
  <div>
    <div
      class="backdrop opened"
      data-testid="backdrop"
    />
    <div
      class="modal opened"
      data-testid="modal"
    >
      <div
        class="header"
      >
        kigasvo
        <div
          class="close"
          data-testid="modal-close"
        >
          x
        </div>
      </div>
      <div
        class="content"
      >
        <img
          src="http://ololacin.vn/hab"
        />
      </div>
    </div>
  </div>
</body>
`;
